### PR TITLE
Machine ID + GitHub Actions bot UI

### PR DESCRIPTION
**Pull request by mcbattirola**
Moved this back to draft since the team will discuss some of the machine ID requirements, which might impact this PR.

It can be reviewed, but shouldn't be merged for now.

_____

TODO:

- [ ] Limit width of the code block (yaml file)
- [ ] Add buttons to download and copy workflow yaml file

_____

This PR adds a new integration flow: Machine ID/GitHub Actions.

We expect new guided flows to be implemented in the future, so I've moved existing Machine ID integration files to a new directory called `MachinId`, and added the github flow code inside a directory named `Github`. This way, future flows can have a dir inside `MachineId`, while reused code goes into `Shared` (same structure as the `Discovery` code). Please let me know what you think.

Demo:

https://github.com/gravitational/teleport/assets/8400114/086886f1-9163-463e-bac0-0a7cfc541b94

Designs:

https://www.figma.com/file/v6GunK50D2VC7w7I2FBDNf/Management?type=design&node-id=3015-32180&mode=design&t=RtWwA6wFPBOlQWQy-0

Contributes to https://github.com/gravitational/teleport/issues/34300

### CODE REVIEW

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -909,7 +909,8 @@ func (h *Handler) bindDefaultEndpoints() {
 	// Channel can contain "/", hence the use of a catch-all parameter
 	h.GET("/webapi/automaticupgrades/channel/*request", h.WithUnauthenticatedHighLimiter(h.automaticUpgrades))
 
-	// Create Machine ID bots
+	// Machine ID bots
+	h.GET("/webapi/sites/:site/machine-id/bot", h.WithClusterAuth(h.getBot))
```
Comment (strideynet) [inline lib/web/apiserver.go:913]: Should this be `/webapi/sites/:site/machine-id/bot/:name` so we don't conflict with a list endpoint?

File: web/packages/teleport/src/Integrations/Enroll/MachineId/GitHub/useGitHubFlow.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+function getRoleYaml(botName: string, labels: ResourceLabel[], login): string {
+  const labelsStanza = labels.map(
+    label => `'${label.name}': '${label.value}'\n`
+  );
+  const timestamp = new Date().getTime();
+
+  return `kind: role
+metadata:
+  name: bot-${botName}-role-${timestamp}
```
Comment (mcbattirola) [inline web/packages/teleport/src/Integrations/Enroll/MachineId/GitHub/useGitHubFlow.tsx:219]: Any suggestions for the role name?

Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/MachineId/GitHub/useGitHubFlow.tsx:219]: We could ask folks to name the integration and then use this in the name of the bot, role and token?

**Comment by mcbattirola** [comment]
Closing this. I'll open a new PR out of a new branch in the future.

**Pull request closed without merging**